internal/controllers/machine/machine_controller.go

```diff
@@ -412,10 +410,16 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Clu
 	}
 
 	if ok, err := r.reconcileDeleteInfrastructure(ctx, m); !ok || err != nil {
+		if err == nil {
+			log.Info("Waiting for infrastructure to be deleted", "Infrastructure", klog.KRef("", m.Spec.InfrastructureRef.Name))
+		}
 		return ctrl.Result{}, err
 	}
```

Comment:
```suggestion
infrastructureDeleted, err := r.reconcileDeleteInfrastructure(ctx, m)
	if err != nil {
		return ctrl.Result{}, err
	}
	if !infrastructureDeleted {
		log.Info("Waiting for infrastructure to be deleted", m.Spec.InfrastructureRef.Kind, klog.KRef(m.Spec.InfrastructureRef.Namespace, m.Spec.InfrastructureRef.Name))
		return ctrl.Result{}, nil
	}
```

WDYT about something like this instead? Just to make it very clear and readable.

Reply:
@schrej Can we also use the actual Kind as key + the Namespace in klog.KRef?

Reply:
Ah, sorry, missed that. Makes sense, was copy pasting from that log line related to Node, but that's different of course.